src/main/java/org/cloudfoundry/promregator/lite/cfaccessor/SimpleCFAccessorImpl.kt

```diff
@@ -0,0 +1,511 @@
+//                connectionContext.dispose()
+//            }
+//        }
+//        val proxyConfiguration = proxyConfiguration()
+//        val connectionContext = connectionContext(proxyConfiguration)
+//        val tokenProvider = tokenProvider()
+//        cloudFoundryClient = cloudFoundryClient(connectionContext, tokenProvider)
+//    } catch (e: ConfigurationException) {
+//        log.error("Restarting Cloud Foundry Client failed due to Configuration Exception raised", e)
+//    }
+
+    suspend fun retrieveAllOrgIds(api: String): List<OrganizationResourceV2> {
+        return cfQueryV2<OrganizationResourceV2, ListOrganizationsResponseV2, ListOrganizationsRequestV2>(api, ORG) {
+            request { page ->
+                ListOrganizationsRequestV2.builder()
+                        .orderDirection(OrderDirection.ASCENDING)
+                        .resultsPerPage(100)
+                        .page(page)
+                        .build()
+            }
+            query { organizations().list(it) }
+        }
+    }
+
+    /* (non-Javadoc)
+	 * @see org.cloudfoundry.promregator.cfaccessor.CFAccessor#retrieveSpaceIdsInOrg(java.lang.String)
+	 */
+    suspend fun retrieveSpaceIdsInOrg(api: String, orgId: String): List<SpaceResourceV2> {
+        return cfQueryV2<SpaceResourceV2, ListSpacesResponseV2, ListSpacesRequestV2>(api, SPACE_IN_ORG) {
+            request { pageNumber ->
+                ListSpacesRequestV2.builder()
+                        .organizationId(orgId)
+                        .orderDirection(OrderDirection.ASCENDING)
+                        .resultsPerPage(100)
+                        .page(pageNumber)
+                        .build()
+            }
+            query { spaces().list(it) }
+        }
+    }
+
+    /* (non-Javadoc)
+	 * @see org.cloudfoundry.promregator.cfaccessor.CFAccessor#retrieveAllApplicationIdsInSpace(java.lang.String, java.lang.String)
+	 */
+    suspend fun retrieveAllApplicationIdsInSpace(api: String, orgId: String, spaceId: String): List<ApplicationResourceV2> {
+        return cfQueryV2<ApplicationResourceV2, ListApplicationsResponseV2, ListApplicationsRequestV2>(api, ALL_APPS_IN_SPACE) {
+            request { pageNumber ->
+                ListApplicationsRequestV2.builder()
+                        .organizationId(orgId)
+                        .spaceId(spaceId)
+                        .orderDirection(OrderDirection.ASCENDING)
+                        .resultsPerPage(100)
+                        .page(pageNumber)
+                        .build()
+            }
+            query { applicationsV2().list(it) }
+        }
+    }
+
+    suspend fun retrieveSpaceSummary(api: String, spaceId: String): GetSpaceSummaryResponse? {
+        // Note that GetSpaceSummaryRequest is not paginated
+        val request = GetSpaceSummaryRequest.builder().spaceId(spaceId).build()
+        return try {
+            cloudFoundryClients[api]?.spaces()?.getSummary(request)?.awaitSingle()
```

Comment:
Note the `awaitSingle` call that converts the reactive call into a suspending coroutines call. Also error handling is managed by simple try/catch constructs.